### PR TITLE
Github Actions:  Fix Build Workflow, Improve Build Output

### PR DESCRIPTION
**Pull request by Jake-Carter**
## Overview
- Fix build workflow not checking out the correct ref
- Add explicit tests for rebuilding PeriphDrivers from scratch
- Improve build output clarity (colors!)
- Fix `Library_Use` build error for `BOARD=Aud01_RevA`

### CODE REVIEW

**Review by Jacob-Scheiffler — COMMENTED** [review]
I'm not a python or workflow wiz so this might be a dumb question, but is it building all projects for all boards every time a commit is pushed? If so, consider only rebuilding for affected projects/micros to reduce runtime.

**Review by sihyung-maxim — APPROVED** [review]

**Comment by sihyung-maxim** [comment]
> I'm not a python or workflow wiz so this might be a dumb question, but is it building all projects for all boards every time a commit is pushed? If so, consider only rebuilding for affected projects/micros to reduce runtime.

I won't merge in this PR so Jacob's question can get attention.

File: Examples/MAX78000/Library_Use/project.mk [hunk 1]
```diff
@@ -13,3 +13,7 @@
 IPATH += lib/include
 PROJ_LDFLAGS += -Llib
 PROJ_LIBS += myLib
+
+ifeq ($(BOARD),Aud01_RevA)
+$(error ERR_NOTSUPPORTED: This project is not supported for the Audio board)
+endif
```
[the remark this answers is not in the document]
Comment (Jake-Carter) [reply Examples/MAX78000/Library_Use/project.mk:19]: No problem at all, we hadn't merged in the latest build tests for your PR so it was easy to miss

**Comment by Jake-Carter** [comment]
> I'm not a python or workflow wiz so this might be a dumb question, but is it building all projects for all boards every time a commit is pushed? If so, consider only rebuilding for affected projects/micros to reduce runtime.

Yes.  It's somewhat involved to isolate changes to specific micros because of how we share driver files across them.  Script complexity vs benefit isn't worth it right now but if we run out of workflow minutes I can address it later

**Review by Jacob-Scheiffler — APPROVED** [review]

**Pull request merged**